proof/crefine/RISCV64/ADT_C.thy

```diff
@@ -1143,21 +1143,21 @@ lemma map_to_cnes_eq:
   apply (clarsimp simp: projectKO_opt_cte split: kernel_object.splits)
   done
 
-lemma ksPSpace_valid_pspace_scRefs_nonzero:
-  "\<lbrakk>\<exists>s. ksPSpace s = ah \<and> valid_pspace' s; map_to_scs ah p = Some sc\<rbrakk> \<Longrightarrow>
+lemma map_to_scs_Some_scRefs_nonzero:
+  "\<lbrakk>map_to_scs (ksPSpace s) p = Some sc; valid_objs' s; no_0_obj' s\<rbrakk> \<Longrightarrow>
```

Comment:
It's kinda sad that we have to give up on the heap abstraction that was going on in ADT_C before because we need more and more invariants (this is not only in MCS, we also needed to do some of that in AARCH64 on master). I guess we could phrase a new `no_0_objs` that takes `ksPSpace s` instead of `s` here, but that does sound out of proportion to what we want to achieve.

Long story short, it's Ok to do this here, but it seems the projection approach that started here doesn't fully carry through without more work.